nexus/pkg/artifacts/downloader.go

```diff
@@ -25,96 +21,70 @@ type ArtifactDownloader struct {
 	// Input
-	QualifiedName          string
-	DownloadRoot           *string
+	ArtifactID             string
+	DownloadRoot           string
 	Recursive              *bool
 	AllowMissingReferences *bool
 	// Properties
-	UpstreamArtifacts   []gql.ArtifactByIDArtifact
-	DefaultDownloadRoot *string
+	UpstreamArtifacts []gql.ArtifactByIDArtifact
 }
 
 func NewArtifactDownloader(
 	ctx context.Context,
 	graphQLClient graphql.Client,
 	downloadManager *filetransfer.FileTransferManager,
-	qualifiedName string,
-	downloadRoot *string,
+	artifactID string,
+	downloadRoot string,
 	recursive *bool,
 	allowMissingReferences *bool,
 ) *ArtifactDownloader {
 	return &ArtifactDownloader{
 		Ctx:                    ctx,
 		GraphqlClient:          graphQLClient,
 		DownloadManager:        downloadManager,
-		QualifiedName:          qualifiedName,
+		ArtifactID:             artifactID,
 		DownloadRoot:           downloadRoot,
 		Recursive:              recursive,
 		AllowMissingReferences: allowMissingReferences,
 		UpstreamArtifacts:      []gql.ArtifactByIDArtifact{},
-		DefaultDownloadRoot:    nil,
 	}
 }
 
-func (ad *ArtifactDownloader) getArtifact() (attrs gql.ArtifactByNameProjectArtifact, rerr error) {
-	entityName, projectName, artifactName, err := parseArtifactQualifiedName(ad.QualifiedName)
-	if err != nil {
-		return gql.ArtifactByNameProjectArtifact{}, err
-	}
-	response, err := gql.ArtifactByName(
+func (ad *ArtifactDownloader) getArtifact() (attrs gql.ArtifactByIDArtifact, rerr error) {
+	response, err := gql.ArtifactByID(
 		ad.Ctx,
 		ad.GraphqlClient,
-		entityName,
-		projectName,
-		artifactName,
+		ad.ArtifactID,
 	)
 	if err != nil {
-		return gql.ArtifactByNameProjectArtifact{}, err
+		return gql.ArtifactByIDArtifact{}, err
 	}
-	return *response.GetProject().GetArtifact(), nil
-}
-
-func loadManifestFromURL(url string) (Manifest, error) {
-	resp, err := http.Get(url)
-	if err != nil {
-		return Manifest{}, err
+	artifact := response.GetArtifact()
+	if artifact == nil {
+		return gql.ArtifactByIDArtifact{}, fmt.Errorf("could not access artifact")
 	}
-	defer resp.Body.Close()
-	manifest := Manifest{}
-	if resp.StatusCode == http.StatusOK {
-		body, err := io.ReadAll(resp.Body)
-		if err != nil {
-			return Manifest{}, fmt.Errorf("error reading response body: %v", err)
-		}
-		err = json.Unmarshal(body, &manifest)
-		if err != nil {
-			return Manifest{}, nil
-		}
-	} else {
-		return Manifest{}, fmt.Errorf("request to get manifest from url failed with status code: %d", resp.StatusCode)
-	}
-	return manifest, nil
+	return *artifact, nil
 }
 
-func (ad *ArtifactDownloader) getArtifactManifest() (artifactManifest Manifest, rerr error) {
-	entityName, projectName, artifactName, err := parseArtifactQualifiedName(ad.QualifiedName)
-	if err != nil {
-		return Manifest{}, err
-	}
+func (ad *ArtifactDownloader) getArtifactManifest(artifact gql.ArtifactByIDArtifact) (manifest Manifest, rerr error) {
 	response, err := gql.ArtifactManifest(
```

Comment:
Nit: we can fetch the artifact by id in this query, and then we won't need `ArtifactByID` at all.